internal/pkg/task/default_runner.go

```diff
@@ -0,0 +1,69 @@
+package task
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ec2"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ecs"
+)
+
+// DefaultRunner can run an Amazon ECS task in the default VPC and the default cluster.
+type DefaultVPCRunner struct {
+	// Count of the tasks to be launched.
+	Count     int
+	// Group Name fo the tasks that use the same task definition.
+	GroupName string
+
+	VPCGetter     VPCGetter
+	ClusterGetter DefaultClusterGetter
+	Starter       TaskRunner
+}
+
+// Run runs tasks in default cluster and default subnets, and returns the task ARNs.
+func (r *DefaultVPCRunner) Run() ([]string, error) {
+	if err:= r.validateDependencies(); err != nil {
+		return nil, err
+	}
+
+	cluster, err := r.ClusterGetter.DefaultCluster()
+	if err != nil {
+		return nil, fmt.Errorf("get default cluster: %w", err)
+	}
+
+	subnets, err := r.VPCGetter.GetSubnetIDs(ec2.FilterForDefaultVPCSubnets)
+	if err != nil {
+		return nil, fmt.Errorf("get default subnet IDs: %w", err)
+	}
+	if len(subnets) == 0 {
+		return nil, errNoSubnetFound
+	}
+
+	arns, err := r.Starter.RunTask(ecs.RunTaskInput{
+		Cluster:        cluster,
+		Count:          r.Count,
+		Subnets:        subnets,
+		TaskFamilyName: taskFamilyName(r.GroupName),
+		StartedBy:      startedBy,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("run task %s: %w", r.GroupName, err)
+	}
```

Comment:
If the task fails due to a missing NAT/Internet gateway, is there an error we can detect and provide the user with a better messaging so that they don't need to do all the research you did to figure out what was wrong?

Reply:
If the task fails to start due to a missing NAT/Internet gateway, it won't be able to pull the container image, which results in a `ResourceNotReady` error returned from the waiter (the [doc](https://docs.aws.amazon.com/sdk-for-go/api/) says `ResourceNotReady` is returned when the waiter's max attempts have been exhausted, but from my observation it was also returned right after the container fails to pull image). Anyways, it is very generic, so i'm not sure if we could diagnose anything from the error. Maybe we could suggest the user to take a look at the console.

Reply:
Sounds good! :)